### ISSUE TITLE
[Question] Isn't Gosu.record intended for taking screenshots?

### ISSUE THREAD

**Issue by akicho8** [gosu/gosu#672]
I wanted to take screenshots of the screen using the internal functions of Gosu, but I'm having trouble using the record method. Here is the code I got after some trial and error, advised by ChatGPT.

```ruby
require "gosu"

class App < Gosu::Window
  def draw
    draw_rect(0, 0, 32, 32, Gosu::Color::BLUE)
  end

  def button_down(id)
    if id == Gosu::KB_SPACE
      Gosu.record(width, height) { draw }.save("screenshot.png")
    end
  end

  new(320, 240).show
end
```

However, when I run this and press space, I get the following error:

> `save': Gosu::Graphics can only be drawn to while rendering (RuntimeError)

Is it possible that the record method is meant for generating its own new texture, and is not suitable for simply wanting a screenshot of the screen?

**Comment by cyberarm** [comment]
Try replacing `record` with `render`

See the method docs:
https://rubydoc.info/gems/gosu/Gosu#record-class_method

**Comment by jlnr** [comment]
Yes, `render` should avoid this issue. However, I am not sure why it doesn't work with `record` (just less efficient), and I'd like to keep this issue as a reminder to look into it.

**Comment by akicho8** [comment]
Changing `record` to `render` fixed the issue and everything is now working perfectly! This experience has taught me the importance of carefully reading the documentation in the future. Thank you so much for your advice.
